resources/test/fixtures/pylint/consider_merging_isinstance.py

```diff
@@ -0,0 +1,37 @@
+"""Checks use of consider-merging-isinstance"""
+# pylint:disable=line-too-long, simplifiable-condition
+
+
+def isinstances():
+    "Examples of isinstances"
+    var = range(10)
+
+    # merged
+    if isinstance(var[1], (int, float)):
+        pass
+    result = isinstance(var[2], (int, float))
+
+    # not merged
+    if isinstance(var[3], int) or isinstance(var[3], float) or isinstance(var[3], list) and True:  # [consider-merging-isinstance]
+        pass
+    result = isinstance(var[4], int) or isinstance(var[4], float) or isinstance(var[5], list) and False  # [consider-merging-isinstance]
+
+    result = isinstance(var[5], int) or True or isinstance(var[5], float)  # [consider-merging-isinstance]
+
```

Comment:
<img width="927" alt="image" src="https://user-images.githubusercontent.com/17039389/205432886-b54be501-b7c3-479a-a3d1-774732fb358a.png">

I'm not sure if this is intended, but the start location of a `BoolOp` expression points to the start location of `or`.

Reply:
RustPython's code:

https://github.com/RustPython/RustPython/blob/18af44b7c7410c3448df1f3b080bd358e586d265/compiler/parser/python.lalrpop#L758

```
OrTest: ast::Expr = {
    <e1:AndTest> <location:@L> <e2:("or" AndTest)*> <end_location:@R> => {
        if e2.is_empty() {
            e1
        } else {
            let mut values = vec![e1];
            values.extend(e2.into_iter().map(|e| e.1));
            ast::Expr {
                location,
                end_location: Some(end_location),
                custom: (),
                node: ast::ExprKind::BoolOp { op: ast::Boolop::Or, values }
            }
        }
    },
};
```

Reply:
Moving `<location:@L>` before `<e1:AndTest>` seems to work?

```
OrTest: ast::Expr = {
    <location:@L> <e1:AndTest> <e2:("or" AndTest)*> <end_location:@R> => {
    ^^^^^^^^^^^^^
        if e2.is_empty() {
            e1
        } else {
            let mut values = vec![e1];
            values.extend(e2.into_iter().map(|e| e.1));
            ast::Expr {
                location,
                end_location: Some(end_location),
                custom: (),
                node: ast::ExprKind::BoolOp { op: ast::Boolop::Or, values }
            }
        }
    },
};
```


<img width="927" alt="image" src="https://user-images.githubusercontent.com/17039389/205433820-5679cb35-918d-4217-842b-d4bda29e03c1.png">

Reply:
```python
import ast

# Visitor to print out node locations
class V(ast.NodeVisitor):
    def visit_BoolOp(self, node: ast.BoolOp) -> None:
        print(
            node.__class__.__name__,
            (node.lineno, node.col_offset),
            (node.end_lineno, node.end_col_offset),
        )
        self.generic_visit(node)

V().visit(ast.parse("a or b or c"))
```

prints out:

```
BoolOp (1, 0) (1, 11)
```

Reply:
Oh cool. Do you want to submit an upstream PR for this? Or I can.

Reply:
Filed https://github.com/RustPython/RustPython/pull/4306.